File: 424-longest-repeating-character-replacement/solution.cpp
// Runtime: 7ms, Beats 80.97% of users with C++
// Memory: 8.34MB, Beats 27.13% of users with C++

class Solution {
public:
    int characterReplacement(string s, int k) {
        int n=s.length();
        int i=0;
        int res=0;
        int maxf=0;
        vector<int> count(26,0);
        for(int j=0;j<n;j++)
        {
            count[s[j]-'A']++;
            maxf=max(maxf,count[s[j]-'A']);
            while((j-i+1)-maxf>k)
            {
                count[s[i]-'A']--;
                i++;
            }
            res=max(res,j-i+1);
        }
        return res;
    }
};

File: 424-longest-repeating-character-replacement/solution.java
// Runtime: 10ms, Beats 66.81% of users with Java
// Memory: 42.75MB, Beats 43.33% of users with Java

class Solution {
    public int characterReplacement(String s, int k) {
        int i=0;
        int n=s.length();
        int res=0;
        int maxf=0;
        int [] count=new int[26];
        for(int j=0;j<n;j++)
        {
            count[s.charAt(j)-'A']++;
            maxf=Math.max(maxf,count[s.charAt(j)-'A']);
            if(j-i+1-maxf>k)
            {
                count[s.charAt(i)-'A']--;
                i++;
            }
            res=Math.max(res,j-i+1);
        }
        return res;
    }
}